packages/devextreme/js/ui/tree_view/ui.tree_view.base.js

```diff
@@ -284,8 +284,17 @@ const TreeViewBase = HierarchicalCollectionWidget.inherit({
                 }
                 break;
             case 'showCheckBoxesMode':
-                this._checkBoxModeChange(value, previousValue);
-                this._invalidate();
+                {
+                    const searchEnabled = this.option('searchEnabled');
+                    const prevSelectAllEnabled = this._selectAllEnabled(previousValue);
```

Comment:
[naming, minor]
Let's not use abbreviations. I understand that `prev` is a basic one, but I prefer have a common rule :)
```suggestion
                    const previousSelectAllEnabled = this._selectAllEnabled(previousValue);
```